src/eko/anomalous_dimensions/nnlo.py

```diff
@@ -0,0 +1,593 @@
+
+    Implements Eq. (3.7) of :cite:`Moch:2004pa`.
+
+    Parameters
+    ----------
+        n : complex
+            Mellin moment
+        nf : int
+            Number of active flavors
+        sx : np array
+            List of harmonic sums
+
+    Returns
+    -------
+        gamma_nsp_2 : complex
+            Next-to-next-to-leading-order singlet-like non-singlet anomalous dimension
+            :math:`\\gamma_{ns,+}^{(2)}(N)`
+    """
+    S1 = sx[0]
+    S2 = sx[1]
+    S3 = sx[2]
+
+    E1 = S1 / n ** 2 + (S2 - zeta2) / n
+    E2 = 2.0 * (-S1 / n ** 3 + (zeta2 - S2) / n ** 2 - (S3 - zeta3) / n)
+
+    pp2 = (
+        -1174.898 * (S1 - 1.0 / n)
+        + 1295.384
+        - 714.1 * S1 / n
+        - 522.1 / (n + 3.0)
+        + 243.6 / (n + 2.0)
+        - 3135.0 / (n + 1.0)
+        + 1641.1 / n
+        - 1258.0 / n ** 2
+        + 294.9 * 2.0 / n ** 3
+        - 800 / 27.0 * 6.0 / n ** 4
+        + 128 / 81.0 * 24.0 / n ** 5
+        + 563.9 * E1
+        + 256.8 * E2
+    )
+
+    pp2_nf = (
+        +183.187 * (S1 - 1.0 / n)
+        - 173.924
+        + 5120 / 81.0 * S1 / n
+        + 44.79 / (n + 3.0)
+        + 72.94 / (n + 2.0)
+        + 381.1 / (n + 1.0)
+        - 197.0 / n
+        + 152.6 / n ** 2
+        - 2608.0 / 81.0 * 2.0 / n ** 3
+        + 192.0 / 81.0 * 6.0 / n ** 4
+        - 56.66 * E1
+        + 1.497 * 6.0 / (n + 1.0) ** 4
+    )
+
+    pf2_nfnf = (
+        -(
+            17.0 / 72.0
+            - 2.0 / 27.0 * S1
+            - 10.0 / 27.0 * S2
+            + 2.0 / 9.0 * S3
+            - (12.0 * n ** 4 + 2.0 * n ** 3 - 12.0 * n ** 2 - 2.0 * n + 3.0)
+            / (27.0 * n ** 3 * (n + 1.0) ** 3)
+        )
+        * 32.0
+        / 3.0
+    )
+
+    result = pp2 + nf * pp2_nf + nf ** 2 * pf2_nfnf
+    return result
+
+
+@nb.njit("c16(c16,u1,c16[:])", cache=True)
+def gamma_nsv_2(n, nf: int, sx):
+    """
+    Computes the next-to-next-to-leading-order valence non-singlet anomalous dimension.
+
+    Implements Eq. (3.9) of :cite:`Moch:2004pa`.
+
+    Parameters
+    ----------
+        n : complex
+            Mellin moment
+        nf : int
+            Number of active flavors
+        sx : np array
+            List of harmonic sums
+
+    Returns
+    -------
+        gamma_nsv_2 : complex
+            Next-to-next-to-leading-order valence non-singlet anomalous dimension
+            :math:`\\gamma_{ns,v}^{(2)}(N)`
+    """
+    S1 = sx[0]
+    S2 = sx[1]
+    S3 = sx[2]
+
+    E1 = S1 / n ** 2 + (S2 - zeta2) / n
+    E2 = 2.0 * (-S1 / n ** 3 + (zeta2 - S2) / n ** 2 - (S3 - zeta3) / n)
+    B11 = -(S1 + 1.0 / (n + 1.0)) / (n + 1.0)
+    B12 = -(S1 + 1.0 / (n + 1.0) + 1.0 / (n + 2.0)) / (n + 2.0)
+
+    # with special care for the first moment of x^-1 ln(1-x)
+    if abs(n.imag) < 1.0e-5 and abs(n - 1.0) < 1.0e-5:
+        B1M = -zeta2
+    else:
+        B1M = -(S1 - 1.0 / n) / (n - 1.0)  # pragma: no cover
```

Comment:
To fix this: maybe can you upload your MMa script for the NNLO benchmark in the wiki (for lack of a better place at the moment) and we can add an reference value for N=2?

Reply:
Okay I can add such test!